frameworks.qmd

```diff
@@ -1,119 +1,1785 @@
+platforms to allow users to construct machine learning models that can
+integrate to their ecosystem.
+
+These steps to remove barrier to entry continue to democratize machine
+learning and make it easier to access for beginners and simplify
+workflow for experts.
+
+### Advanced Learning Methods
+
+##### Transfer Learning
+
+Transfer learning is the practice of using knowledge gained from a
+pretrained model to train and improve performance of a model that is for
+a different task. For example, datasets that have been trained on
+ImageNet datasets such as MobileNet and ResNet can help classify other
+image datasets. To do so, one may freeze the pretrained model, utilizing
+it as a feature extractor to train a much smaller model that is built on
+top of the feature extraction. One can also fine tune the entire model
+to fit the new task. Transfer learning has a series of challenges, in
+that the modified model may not be able to conduct its original tasks
+after transfer learning. Papers such as [["Learning without
+Forgetting"]{.underline}](https://browse.arxiv.org/pdf/1606.09282.pdf)
+paper aims to address these challenges and have been implemented in
+modern machine learning platforms.
+
+##### Federated Learning
 
-## Embedded AI Constraints
+Consider the problem of labeling items that are present in a photo from
+personal devices. One may consider moving the image data from the
+devices to a central server, where a single model will train Using these
+image data provided by the devices. However, this presents many
+potential challenges. First, with many devices one needs a massive
+network infrastructure to move and store data from these devices to a
+central location. With the number of devices that are present today this
+is often not feasible, and very costly. Furthermore, there are privacy
+challenges associated with moving personal data, such as Photos central
+servers.
 
-Explanation: Describe the constraints of embedded systems, referring to the previous chapters, and remind readers about the challenges and why we need to consider creating lean and efficient solutions.
+[[Federated learning]{.underline}](https://arxiv.org/abs/1602.05629) is
+a form of distributed computing that resolves these issues by
+distributing the models into personal devices for them to be trained on
+device. At the beginning, a base global model is trained on a central
+server to be distributed to all devices. Using this base model, the
+devices individually compute the gradients and send them back to the
+central hub. Intuitively this is the transfer of model parameters
+instead of the data itself. This innovative approach allows the model to
+be trained with many different datasets (which, in our example, would be
+the set of images that are on personal devices), without the need to
+transfer a large amount of potentially sensitive data. However,
+federated learning also comes with a series of challenges.
```

Comment:
Great paragraph on Federated Learning! I think a graph to illustrate how federated learning works might also be cool.